.github/workflows/pr_push.yml

```
@@ -14,50 +14,87 @@ concurrency:
 
 permissions:
   contents: read
+  packages: read
 
 jobs:
   CodeChecks:
     uses: ./.github/workflows/reusable_checks.yml
   DocsBuild:
     uses: ./.github/workflows/reusable_docs_build.yml
+  DetectChanges:
+    runs-on: ubuntu-latest
+    outputs:
+      changed_files: ${{ steps.changed-files.outputs.all_changed_files }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          fetch-depth: 0
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v45.0.3
+
+      - name: List all changed files
+        env:
+          ALL_CHANGED_FILES: ${{ steps.changed-files.outputs.all_changed_files }}
+        run: |
+          echo "Changed files: ${{ steps.changed-files.outputs.all_changed_files }}"
+  BuildDockers:
+    if: ${{ contains(join(needs.DetectChanges.outputs.changed_files, ' '), '.github/docker/') }}
+    needs: [DetectChanges]
+    permissions:
+      contents: read
+      packages: write
+    secrets: inherit
+    uses: ./.github/workflows/build_ci_container.yml
   FastBuild:
-    name: Fast builds
-    needs: [CodeChecks, DocsBuild]
+    if: always() && (needs.BuildDockers.result == 'skipped' || needs.BuildDockers.result == 'success')
+    needs: [CodeChecks, DocsBuild, BuildDockers]
     uses: ./.github/workflows/reusable_fast.yml
   Build:
     name: Basic builds
+    if: always() && (needs.FastBuild.result == 'success')
```

Is it needed? Failing FastBuild shall block dependent workflows anyway. I think that there is no scenario where FastBuild is skipped either.

---

yes, without it the job can't be picked up as here https://github.com/rbanka1/unified-memory-framework/actions/runs/13564232637/workflow